apis/Google.Cloud.EntityFrameworkCore.Spanner/Google.Cloud.EntityFrameworkCore.Spanner.IntegrationTests/TestDatabaseFixture.cs

```diff
@@ -0,0 +1,189 @@
+                  BytesValue          BYTES(MAX),
+                  TimestampValue      TIMESTAMP,
+                  DateValue           DATE,
+                  BoolArrayValue      ARRAY<BOOL>,
+                  Int64ArrayValue     ARRAY<INT64>,
+                  Float64ArrayValue   ARRAY<FLOAT64>,
+                  StringArrayValue    ARRAY<STRING(MAX)>,
+                  BytesArrayValue     ARRAY<BYTES(MAX)>,
+                  TimestampArrayValue ARRAY<TIMESTAMP>,
+                  DateArrayValue      ARRAY<DATE>,
+                ) PRIMARY KEY (K)";
+
+            await ExecuteDdlAsync(typeTable);
+        }
+
+        private async Task CreateTableAsync()
+        {
+            string createTableStatement = $@"CREATE TABLE {TestTable} (
+                                            Key                STRING(MAX) NOT NULL,
+                                            StringValue        STRING(MAX),
+                                          ) PRIMARY KEY (Key)";
+
+            var index1 = "CREATE INDEX TestTableByValue ON TestTable(StringValue)";
+            var index2 = "CREATE INDEX TestTableByValueDesc ON TestTable(StringValue DESC)";
+
+            await ExecuteDdlAsync(createTableStatement);
+            await ExecuteDdlAsync(index1);
+            await ExecuteDdlAsync(index2);
+        }
+
+        private async Task FillSampleData(string testTable)
+        {
+            using (var connection = new SpannerConnection(ConnectionString))
+            {
+                await connection.OpenAsync();
+                using (var tx = await connection.BeginTransactionAsync())
+                {
+                    var cmd = connection.CreateInsertCommand(
+                        testTable,
+                        new SpannerParameterCollection
+                        {
+                            {"Key", SpannerDbType.String},
+                            {"StringValue", SpannerDbType.String}
+                        });
+                    cmd.Transaction = tx;
+
+                    for (var i = 0; i < TestTableRowCount; ++i)
+                    {
+                        cmd.Parameters["Key"].Value = "k" + i;
+                        cmd.Parameters["StringValue"].Value = "v" + i;
+                        await cmd.ExecuteNonQueryAsync();
+                    }
+
+                    await tx.CommitAsync();
+                }
+            }
+        }
+
+        public async Task<TestTableContext> CreateContextAsync()
+        {
+            await EnsureTestDatabaseAsync();
+            return new TestTableContext(this);
+        }
+    }
+
+    public class TestTableContext : DbContext
+    {
+        private readonly TestDatabaseFixture _fixture;
+
+        public TestTableContext(TestDatabaseFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        public virtual DbSet<StringEntry> StringTable { get; set; }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
```

Comment:
Yes.  There are usually annotations you need to add to your POCOs to get things working and some of those can be provider specific.  We won't have any -- possibly just one to help with conflict detection and resolution -- but it remains to be seen what level of support we can achieve for other database annotations or if we should even try.
There are also ways to annotate outside of the POCO class which people can do to avoid tying their model to a specific provider.  That's going to be a much more likely path to success with us.